idanalysis/eqparams.py

```diff
@@ -0,0 +1,1294 @@
+        tauy = 2 * energy * t0 / (u0 * jy)
+        taue = 2 * energy * t0 / (u0 * jz)
+
+        return taux, tauy, taue
+
+    def add_id_to_lattice(self, insertion_device):
+        """Add ID to lattice.
+
+        Args:
+            insertion_device (InsertionParams object): Object from class
+             InsertionParams.
+        """
+        self.ids.append(insertion_device)
+
+    def calc_total_effect_on_eqparams(self):
+        """Calculate cumulative effect on equilibrium parameters.
+
+        Returns:
+            numpy 1d array: Horizontal emittance [m rad]
+            numpy 1d array: Vertical emittance [m rad]
+            numpy 1d array: Energy spread
+            numpy 1d array: Horizontal damping time [s]
+            numpy 1d array: Vertical damping time [s]
+            numpy 1d array: Longitudinal damping time [s]
+        """
+        emitx = _np.zeros(len(self.ids) + 1)
+        emity = _np.zeros(len(self.ids) + 1)
+        espread = _np.zeros(len(self.ids) + 1)
+        taux = _np.zeros(len(self.ids) + 1)
+        tauy = _np.zeros(len(self.ids) + 1)
+        taue = _np.zeros(len(self.ids) + 1)
+
+        if self.eq_params_nominal is None:
+            self.get_nominal_model_eqparams()
+        emitx[0] = self.eq_params_nominal.emitx
+        emity[0] = self.eq_params_nominal.emity
+        espread[0] = self.eq_params_nominal.espread0
+        taux[0] = self.eq_params_nominal.taux
+        tauy[0] = self.eq_params_nominal.tauy
+        taue[0] = self.eq_params_nominal.taue
+
+        di2x, di2y = 0, 0
+        di4x, di4y = 0, 0
+        di5x, di5y = 0, 0
+        di3, du0 = 0, 0
+        for i, insertion_device in enumerate(self.ids):
+            if insertion_device.field_profile is None:
+                insertion_device.create_field_profile()
+            di2x_, di2y_ = insertion_device.calc_i2()
+            di4x_, di4y_ = insertion_device.calc_i4()
+            di5x_, di5y_ = insertion_device.calc_i5()
+            di3_, _ = insertion_device.calc_i3()
+            du0_ = 1e-3 * insertion_device.calc_u0() * ECHARGE
+            di2x += di2x_
+            di2y += di2y_
+            di4x += di4x_
+            di4y += di4y_
+            di5x += di5x_
+            di5y += di5y_
+            di3 += di3_
+            du0 += du0_
+            emitx[i + 1], emity[i + 1] = self.calc_id_effect_on_emittance(
+                None, di2x, di2y, di4x, di4y, di5x, di5y
+            )
+            espread[i + 1] = self.calc_id_effect_on_espread(
+                None, di2x, di3, di4x
+            )
+            (
+                taux[i + 1],
+                tauy[i + 1],
+                taue[i + 1],
+            ) = self.calc_id_effect_on_damping_times(None, di2x, di4x, du0)
+        self._emitx = emitx
+        self._emity = emity
+        self._espread = espread
+        self._taux = taux
+        self._tauy = tauy
+        self._taue = taue
+        return emitx, emity, espread, taux, tauy, taue
```

Comment:
Do you really need to return these quantities? If they are stored in object properties I don't see why.